### ISSUE TITLE
Issue triaging bot should add need-triage if the triaging is not complete 

### ISSUE THREAD

**Issue by weshaggard** [Azure/azure-sdk-tools#6479]
As an example see issue https://github.com/Azure/azure-sdk-tools/issues/6474. When it was created it was assigned to someone but didn't have any labels added to it so I would have expected this to add the "needs-triage" label so that triaging could be completed.

For triaging purposes I think we at least need to have some Labels added to the item.

cc @jsquire @JimSuplizio

**Comment by jsquire** [comment]
In the example case, the author of the issue should have assigned labels if they were desired.   

If the person creating the issue has rights to assign a person and chooses to do so, they have assumed responsibility for that issue.   This is also the case when labels are added but a person is not assigned.   Both of these are valid states for an issue.   We do not want automation to assume that it knows better than a human and start adding labels, assignees, or making other changes that potentially overwrite the intent of the author.   

I'll leave this for Jim to weigh in, but my inclination is to "by design" this.

**Comment by weshaggard** [comment]
@kurtzeborn what are your thoughts here?

We always get folks creating issues and somethings they assign or label incompletely. However I certainly think for labels we should always have a label added and if the author added one great but if they didn't we should add the "needs-triage" label because that will trigger it to show up for triage by the triaging team.

**Comment by kurtzeborn** [comment]
We talked about this today.  The "needs-triage" label should be added whenever there are no labels on an issue.  Whether the issue is assigned to someone or not is mostly irrelevant for this rule.  The labels (such as "Central-EngSys") are what cause issues to be automatically included into appropriate GitHub projects.

So the bot should add "needs-triage" to newly created issues (since the last run) which have no labels on them already.  If there are labels on the issue, then the bot can skip adding "needs-triage".

**Comment by jsquire** [comment]
@kurtzeborn:  Respectfully, that's not how the triage process is intended to work.   As the owner of the automation rules and repository process, that's not something that I'm comfortable bringing forward as a generalized rule.  

Conceptually, I have no objections to using a different flow in tools, but we don't have a mechanism for overriding logic.  You would have to turn off the common triage rule and write a custom one that was only enabled here.  IMHO, that's more work and maintenance than is justified but I won't block if you feel strongly about it.

**Comment by weshaggard** [comment]
@jsquire from your perspective do you look at issues in the repo that have no labels applied (regardless if they are assigned or not)? If not when do these issues with no labels get looked at? It seems like there is opportunity for issues to fall through the cracks and not ever show up in any triaging meeting.

**Comment by jsquire** [comment]
@weshaggard: Personally, I do tend to notice when tags are off, as I'm doing multiple triage scans a day.  In most repos, someone will notice the lack of tags and follow-up.  They also appear on the data quality reports run by our PM team.

That said, the person making the assignment must have write permissions to our repository and, as such, is a known partner or team member.  If they assign a person to an issue, that person is assumed to be responsible for that issue.   While we would like people to follow our patterns for tagging, people will often not when opening short-lived issues for tracking and the folks doing repo triage are not expected to clean up the mess.

There is also the issue of the race condition for automation where an issue is created without issues, then in separate updates assignments and tags are done.  The amount of false positive "needs-team-triage" that occur as a result of that has been disruptive to metrics and causing extra work for those doing triage.  I've tuned that rule to consider assignments as well and do not want to reintroduce the noise for a "needs-triage" race condition.

**Comment by weshaggard** [comment]
I agree the race conditions are annoying and we could potentially handle some of that clean-up via a scheduled job if we chose to.

While folks that create an issue as well as assign it do have permissions and could label things, we know not everyone that has permissions knows to label or how to correctly label. If you are OK with the potential of issues dropping through the cracks and the assignee also not correctly routing or responding to the issue then I'll stop pushing on this. For EngSys triaging we will just need to update our triaging of the tools repo to find all issues with no labels or with "needs-triage" label in order to help ensure everything gets looked at by the triage team. I suspect other repos might also need to do something similar.
